src/cross_toml.rs

```diff
@@ -116,6 +111,37 @@ impl CrossToml {
     fn get_target(&self, target: &Target) -> Option<&CrossTargetConfig> {
         self.targets.get(target)
     }
+
+    fn get_string<Get>(&self, target: &Target, get: Get) -> Option<String>
+    where
+        Get: Fn(&CrossTargetConfig) -> &Option<String>,
+    {
+        self.get_target(target).and_then(|t| get(t).clone())
+    }
+
+    fn get_bool<GetBuild, GetTarget>(
+        &self,
+        target: &Target,
+        get_build: GetBuild,
+        get_target: GetTarget,
+    ) -> (Option<bool>, Option<bool>)
+    where
+        GetBuild: Fn(&CrossBuildConfig) -> Option<bool>,
+        GetTarget: Fn(&CrossTargetConfig) -> Option<bool>,
+    {
+        let build = get_build(&self.build);
+        let target = self.get_target(target).and_then(get_target);
+
+        (build, target)
+    }
+
+    fn get_vec<Get>(&self, target: &Target, get: Get) -> Vec<String>
+    where
+        Get: for<'a> Fn(&'a CrossEnvConfig) -> &'a Vec<String>,
+    {
+        self.get_target(target)
+            .map_or(Vec::new(), |t| get(&t.env).clone())
```

Comment:
same here, we could return `&[String]`

Reply:
I've changed all of these to return `&[String]`. The only major change required was then to ensure that the sum had the appropriate lifetimes.

```rust
fn sum_of_env_toml_values<'a>(
    &'a self,
    env_values: Option<Vec<String>>,
    toml_getter: impl FnOnce(&'a CrossToml) -> &'a [String],
) -> Result<Vec<String>> {
    let mut collect = vec![];
    if let Some(mut vars) = env_values {
        collect.append(&mut vars);
    } else if let Some(toml_values) = self.toml.as_ref().map(toml_getter) {
        collect.extend(toml_values.iter().cloned());
    }

    Ok(collect)
}
```